src/core/meshcop/tcat_agent.cpp

```diff
@@ -540,6 +547,97 @@ Error TcatAgent::HandlePing(const Message &aIncomingMessage,
     return error;
 }
 
+Error TcatAgent::HandleGetNetworkName(Message &aOutgoingMessage)
+{
+    Error       error  = kErrorNone;
+    const char *name   = otThreadGetNetworkName(&GetInstance());
+    uint16_t    length = static_cast<uint16_t>(strlen(name));
+    ot::Tlv     tlv;
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+    VerifyOrExit(length > 0, error = kErrorInvalidState);
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
+    SuccessOrExit(error = aOutgoingMessage.Append(tlv));
+    SuccessOrExit(error = aOutgoingMessage.AppendBytes(name, length));
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetDeviceId(Message &aOutgoingMessage)
+{
+    uint8_t  deviceId[OT_EXT_ADDRESS_SIZE];
+    uint16_t length = 0;
+    ot::Tlv  tlv;
+    Error    error = kErrorNone;
+
+    static_assert(OT_EXT_ADDRESS_SIZE > OT_TCAT_MAX_VENDORID_SIZE, "Size of TCAT vendor id is greater than extaddr.");
+
+    if (mVendorInfo->mDeviceIds != nullptr)
+    {
+        // If array is not null return first set device ID.
+        for (uint8_t i = 0; mVendorInfo->mDeviceIds[i].mDeviceIdType != OT_TCAT_DEVICE_ID_EMPTY; i++)
+        {
+            length = mVendorInfo->mDeviceIds->mDeviceIdLen;
+            memcpy(deviceId, mVendorInfo->mDeviceIds[i].mDeviceId, length);
+            break;
+        }
+    }
+
+    if (length == 0)
+    {
+        // If array is null or contains only empty entries return EUI64.
+        otPlatRadioGetIeeeEui64(&GetInstance(), deviceId);
+        length = OT_EXT_ADDRESS_SIZE;
+    }
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
+    SuccessOrExit(error = aOutgoingMessage.Append(tlv));
+    SuccessOrExit(error = aOutgoingMessage.AppendBytes(deviceId, length));
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetExtPanId(Message &aOutgoingMessage)
+{
+    Error                  error = kErrorNone;
+    const otExtendedPanId *name =
+        otThreadGetExtendedPanId(&GetInstance()); // is it possible to check if i't not set ask during review.
+    uint16_t length = OT_EXT_PAN_ID_SIZE;
+    ot::Tlv  tlv;
+
+    VerifyOrExit(Get<ActiveDatasetManager>().IsCommissioned(), error = kErrorInvalidState);
+
+    tlv.SetType(kTlvResponseWithPayload);
+    tlv.SetLength(static_cast<uint8_t>(length));
+    SuccessOrExit(error = aOutgoingMessage.Append(tlv));
+    SuccessOrExit(error = aOutgoingMessage.AppendBytes(name, length));
+
+exit:
+    return error;
+}
+
+Error TcatAgent::HandleGetProvisioningUrl(Message &aOutgoingMessage)
+{
+    Error    error  = kErrorNone;
+    uint16_t length = static_cast<uint16_t>(strlen(mVendorInfo->mProvisioningUrl));
```

Comment:
Is `mVendorInfo->mProvisioningUrl` guaranteed to be non-null? Should we check?

Reply:
it's verified in here https://github.com/openthread/openthread/blob/main/src/core/meshcop/tcat_agent.cpp#L104
And `strlen` should be null safe and return 0 in such case.